src/patterns/ec2-app.ts

```diff
@@ -468,19 +519,47 @@ export class GuEc2App {
     }
 
     if (!monitoringConfiguration.noMonitoring) {
-      if (monitoringConfiguration.http5xxAlarm) {
+      const {
+        http5xxAlarm,
+        snsTopicName,
+        unhealthyInstancesAlarm,
+        actionsEnabledInCode = false,
+      } = monitoringConfiguration;
+
+      /*
+      Stage is known at runtime if it is a CFN Parameter, e.g. CODE or PROD.
+      Stage is known at compile time for singleton stacks, e.g. INFRA.
+       */
+      const isStageDeterminedAtRuntime = Token.isUnresolved(stage);
```

Comment:
A type guard could have been used here (a la https://github.com/guardian/cdk/pull/1048), however this is slightly simpler.